index.js

```diff
@@ -223,23 +224,27 @@ function create (opts) {
     opened[hex] = old
 
     if (!feed) {
-      old.feed = feed = core.createFeed(key, {
-        storage: storage(path.join(dir, 'data', key.slice(0, 2), key.slice(2) + '.data')),
-        sparse: opts.sparse && isContent
-      })
+      old.feed = feed = hypercore(
+        path.join(dir, 'data', key.slice(0, 2), key.slice(2) + '.data'),
+        key,
+        {sparse: opts.sparse && isContent}
+      )
     }
 
     old.cnt++
-    feed.replicate({stream: stream})
+    var replicator = feed.replicate({stream: stream})
 
+    console.log('first download', feed.firstDownload)
     if (!feed.firstDownload) {
       var downloaded = false
 
       feed.firstDownload = true
       feed.once('download', function () {
+        console.log('downloaded')
         downloaded = true
       })
-      feed.once('download-finished', function () {
+      replicator.on('end', function () {
```

Comment:
This is never getting called, is there a better way to check if the feed has been downloaded?